crates/node/ethstats/src/ethstats.rs

```diff
@@ -0,0 +1,828 @@
+use crate::{
+    connection::ConnWrapper,
+    error::EthStatsError,
+    events::{
+        AuthMsg, BlockMsg, BlockStats, HistoryMsg, LatencyMsg, NodeInfo, NodeStats, PendingMsg,
+        PendingStats, PingMsg, StatsMsg, TxStats, UncleStats,
+    },
+};
+use alloy_consensus::{BlockHeader, Sealable};
+use alloy_primitives::U256;
+use reth_network_api::{NetworkInfo, Peers};
+use reth_primitives_traits::{Block, BlockBody, InMemorySize};
+use reth_provider::{
```

Comment:
I believe we can import these from reth_storage_api?

then we likely dont need the reth-provider dep at all